website/docs/reference/resource-configs/fabric-configs.md

```diff
@@ -3,103 +3,910 @@ title: "Microsoft Fabric DWH configurations"
+
+<File name='bigquery_table.sql'>
+
+```sql
+create table `projectname`.`analytics`.`bigquery_table`
+partition by timestamp_trunc(created_at, day)
+as (
+
+  select
+    user_id,
+    event_name,
+    created_at
+
+  from `analytics`.`events`
+
+)
+```
+
+</File>
+
+</TabItem>
+</Tabs>
+
+#### Partitioning by an "ingestion" date or timestamp
+
+BigQuery supports an [older mechanism of partitioning](https://cloud.google.com/bigquery/docs/partitioned-tables#ingestion_time) based on the time when each row was ingested. While we recommend using the newer and more ergonomic approach to partitioning whenever possible, for very large datasets, there can be some performance improvements to using this older, more mechanistic approach. [Read more about the `insert_overwrite` incremental strategy below](#copying-ingestion-time-partitions).
+
+dbt will always instruct BigQuery to partition your table by the values of the column specified in `partition_by.field`. By configuring your model with `partition_by.time_ingestion_partitioning` set to `True`, dbt will use that column as the input to a `_PARTITIONTIME` pseudocolumn. Unlike with newer column-based partitioning, you must ensure that the values of your partitioning column match exactly the time-based granularity of your partitions.
+
+<Tabs
+  defaultValue="source"
+  values={[
+    { label: 'Source code', value: 'source', },
+    { label: 'Compiled code', value: 'compiled', },
+  ]
+}>
+<TabItem value="source">
+
+<File name='bigquery_table.sql'>
+
+```sql
+{{ config(
+    materialized="incremental",
+    partition_by={
+      "field": "created_date",
+      "data_type": "timestamp",
+      "granularity": "day",
+      "time_ingestion_partitioning": true
+    }
+) }}
+
+select
+  user_id,
+  event_name,
+  created_at,
+  -- values of this column must match the data type + granularity defined above
+  timestamp_trunc(created_at, day) as created_date
+
+from {{ ref('events') }}
+```
+
+</File>
+
+</TabItem>
+<TabItem value="compiled">
+
+<File name='bigquery_table.sql'>
+
+```sql
+create table `projectname`.`analytics`.`bigquery_table` (`user_id` INT64, `event_name` STRING, `created_at` TIMESTAMP)
+partition by timestamp_trunc(_PARTITIONTIME, day);
+
+insert into `projectname`.`analytics`.`bigquery_table` (_partitiontime, `user_id`, `event_name`, `created_at`)
+select created_date as _partitiontime, * EXCEPT(created_date) from (
+    select
+      user_id,
+      event_name,
+      created_at,
+      -- values of this column must match granularity defined above
+      timestamp_trunc(created_at, day) as created_date
+
+    from `projectname`.`analytics`.`events`
+);
+```
+
+</File>
+
+</TabItem>
+</Tabs>
+
+#### Partitioning with integer buckets
+
+If the `data_type` is specified as `int64`, then a `range` key must also
+be provided in the `partition_by` dict. dbt will use the values provided in
+the `range` dict to generate the partitioning clause for the table.
+
+<Tabs
+  defaultValue="source"
```

Comment:
our custom components shouldn't get flagged as typos. is it possible to ignore these or maybe add these to our custom dictionary?